includes/downloads.php

```diff
@@ -23,6 +23,76 @@ function podlove_get_remote_addr() {
 	return $_SERVER['REMOTE_ADDR'];
 }
 
+function ga_track_download($request_id, $media_file, $ua_string, $ptm_context, $ptm_source) {	
+	// GA Tracking
+	$debug_ga = false;
+	$ga_collect_endpoint = 'https://www.google-analytics.com/' . ($debug_ga ? 'debug/' : '') . 'collect';
+	
+	$ga_tracking_id = trim(\Podlove\get_setting('tracking', 'ga'));
+	if (!$ga_tracking_id || $ga_tracking_id === '') {
+		return;
+	}
+	
+	$ga_params = array(
+		// Basics
+		'v' => '1', // version
+		'tid' => $ga_tracking_id, // tracking id
+		'cid' => $request_id, // client id
+		'ua' => $ua_string, // user agent override
+		'uip' => podlove_get_remote_addr(), // IP override
+		'ds' => 'podlove', // data source
+		
+		// We highjack the campaign fields for context/source data.
+		// Source / Medium maps to Podlove context / Podlove source.
+		// This way all Podlove sources can be easily grouped into GA Channels.
+		'cs' => $ptm_context, // campaign source
+		'cm' => $ptm_source, // campaign medium
+		'ci' => $media_file->episode()->number, // campaign id
+		'cn' => $media_file->episode()->title(), // campaign name
+		
+		// Pageview params
+		't' => 'pageview', // hit type
+		'dh' => $_SERVER[HTTP_HOST], // document host
+		'dp' => $_SERVER[REQUEST_URI], // document path
+		'dt' => $media_file->episode()->title(), // document title
+	);
+	
+	$ga_param_fragments = array();
+	array_walk($ga_params, function($item, $key) use(&$ga_param_fragments) {
+		array_push($ga_param_fragments, sprintf('%s=%s', $key, rawurlencode($item)));
+	});
+	
+	$body = implode('&', $ga_param_fragments);
+	$curl = new \Podlove\Http\Curl();
+	$curl->request( $ga_collect_endpoint, array(
+		'method'  => 'POST',
+		'body'    => $body,
+	));
+	
+	if (!$curl->isSuccessful()) {
+		if ($debug_ga) {
+			header('x-ga-debug: http error');
+		}
+		\Podlove\Log::get()->addError('GA Measurement Protocol request failed.');
+	} else {
+		\Podlove\Log::get()->addInfo('GA Measurement Protocol request successful: ' . $body);
```

Comment:
I would prefer `addDebug` in all success cases so the log isn't flooded with tracking requests.